app/src/main/java/com/devindi/wallpaper/home/HomeViewModel.kt

```diff
@@ -4,37 +4,48 @@ import android.arch.lifecycle.ViewModel
 import com.devindi.wallpaper.misc.ReportManager
 import com.devindi.wallpaper.model.SettingsRepo
 import com.devindi.wallpaper.model.analytics.CreateWallpaperEvent
-import com.devindi.wallpaper.model.map.MapAreaManager
-import org.osmdroid.util.BoundingBox
+import com.devindi.wallpaper.model.map.MapImageGenerator
+import com.devindi.wallpaper.settings.model.DIMENSION_HEIGHT
+import com.devindi.wallpaper.settings.model.DIMENSION_WIDTH
+import com.devindi.wallpaper.settings.model.SettingsManager
 import com.google.firebase.perf.FirebasePerformance
+import org.osmdroid.util.GeoPoint
+import timber.log.Timber
 
 class HomeViewModel(
-    private val manager: MapAreaManager,
+    private val imageGenerator: MapImageGenerator,
     private val handler: WallpaperHandler,
     private val reportManager: ReportManager,
-    settings: SettingsRepo
+    settings: SettingsRepo,
+    private val settingsManager: SettingsManager
 ) : ViewModel() {
 
     var currentTileSource = settings.currentMapSource()
 
-    fun createWallpaper(boundingBox: BoundingBox, zoomLevel: Int) {
-        Thread({
+    fun createWallpaper(centerPoint: GeoPoint, zoomLevel: Int) {
+        Timber.d("Center point: ${centerPoint.longitude}, ${centerPoint.latitude} ; zoom = $zoomLevel")
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/devindi/mapwall/blob/63cb55c913b0d690492f74a58d6634a0c41df234/app/src/main/java/com/devindi/wallpaper/home/HomeViewModel.kt#L26"></span>Exceeded max line length (100) (cannot be auto-corrected)</td>
    </tr>
  </tbody>
</table>